scheduler/scheduler.go

```diff
@@ -582,25 +710,23 @@ func (driver *MesosSchedulerDriver) doReliableRegistration(maxBackoff float64) {
 // return true if we should attempt another registration later
 func (driver *MesosSchedulerDriver) registerOnce() (proceed bool) {
 
-	proceed = true
 	var (
 		failover bool
 		pid      *upid.UPID
 	)
-
-	func() {
+	if func() bool {
```

Comment:
Stretching the golang syntax in if-statement epitomizes all I like about the language.  :+1: